### ISSUE TITLE
Centralize "keys"

### ISSUE THREAD

**Issue by harunurhan** [inveniosoftware-contrib/ng2-json-editor#272]
This is probably the best solution for solving some issues and features request that we have right now, also it'll reduce the computation being done for sorting and filtering because even if it's not necessary sometimes the pipes are executed anyway.

This will remove the pipes, `AddAlwaysShowFields`, `AddAlwaysShowToggles`, `FilterAndSortBySchema`, and will run the logic on the `init` to create data structure with all the keys. This keys can be updated later (adding and deleting field) but the order should be kept.

## Issues and things

This issues and things either need to this proposal or they will be affected from it.

### #245 (the 5th subtask)

This one is a bit tricky.

To be able to fix this, I think we'll have to create another `map` which is similar, just with different order. And it should be also synced (update when a key is deleted or added).

### #235 

The actually problem is not the mentioned in the issue but related, it's removing keys of table-item from the json that is being emitted out, if they are set as empty.

### #243 

All logic has to run again when admin mode is enabled/disabled

### #227 

Once keys are centralized in a globally accessible place, excluding `alwaysShow` fields from `AddFieldDropdown` will be solved already. However making them non-deletable is another issue which is not related to this, maybe not even necessary.

### Unifying `AddFieldDropdown` for each element of `ComplexList`

This is not really possible with current software design, this requires the proposal

### `TableList` and `TableItem` special case for keys

This will require component type during creation of all keys, because `TableList` items have all the keys that are present at least one of them.

## API

```typescript
interface KeysStore {
   private keysMap: Immutable.Map<string, Immutable.OrderedSet<string>>;
   
   // called only once and populates keysMap
   buildKeysMap(json: Map<string, any>);
   
    // just returns keysMap.get(path)
   forPath(string: path): Immutable.OrderedSet<string>;
}
```
